MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockRng.h

```diff
@@ -41,8 +41,17 @@ struct MockRng {
     );
 };
 
+MOCK_INTERFACE_DEFINITION (MockRng);
+MOCK_FUNCTION_DEFINITION (MockRng, GetInfo, 3, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockRng, GetRng, 4, EFIAPI);
+
+EFI_RNG_PROTOCOL  RNG_PROTOCOL_INSTANCE = {
+  GetInfo, // EFI_RNG_GET_INFO
+  GetRng   // EFI_RNG_GET_RNG
+};
+
 extern "C" {
-  extern EFI_RNG_PROTOCOL  *gRngProtocol;
+  EFI_RNG_PROTOCOL  *gRngProtocol = &RNG_PROTOCOL_INSTANCE;
 }

```

Comment:
What is the naming convention for this global pointer?  Should it include the keyword "Mock" so the unit test source code it is clear that it is a mock interaction.

Also, what if the unit test code is testing code under test that needs to interact with more then one instance of a protocol/ppi?  If only a single global is created like this, does that make writing tests for multiple protocol instances harder?  System tables (BS, RT, DXE, PEI Services) are all examples where there is at most one instance.  Protocols and PPI can have multiple instances.  An example is the PartitionDxe that interacts with multiple Block I/O and Block I/O 2 Protocol instances:

https://github.com/tianocore/edk2/blob/4de80843a74475aef3ac47127cec0e52c1f5aa46/MdeModulePkg/Universal/Disk/PartitionDxe/Partition.c#L491

Instead of declaring a single global pointer here, should the Google Test unit test code in a Setup() function create one or more table/protocol/ppi instances as required for the code under test within the scope of the Setup() method?

Reply:
Do you mean having this code that's in the header file be copied into every consuming test file?
https://github.com/tianocore/edk2/pull/5950/commits/cd820ce0bb2a505ffd725d01e6a99b260a29ec5a
I was trying to avoid this level of code duplication.

Reply:
Here's an example of instead using a macro for instantiation: https://github.com/VivianNK/mu_basecore/commit/aa84c6eca304e74218c9ff93c280084d02ca6b32

Reply:
Updated all protocol and PPI mocks to remove instantations and instead have a macro that can be used for multiple instances. @mdkinney